lib/tools/ui/widgets/text_entry.dart

```diff
@@ -78,6 +78,10 @@ class TextEntry extends StatelessWidget {
       ),
       validator: (value) {
```

Comment:
Even if it works, I think the validator could be cleaner. I would prefer something like this : 

```suggestion
      validator: (value) {
        if (value == null || value.isEmpty) {
          if (canBeEmpty) {
            return null;
          } else {
            return noValueError;
          }
        }

        if (isInt) {
          final intValue = int.tryParse(value);
          if (intValue == null || intValue < 0) {
            return TextConstants.invalidNumber;
          }
        }

        if (isDouble) {
          final doubleValue = double.tryParse(value.replaceAll(',', '.'));
          if (doubleValue == null || doubleValue < 0) {
            return TextConstants.invalidNumber;
          }
        }

        validator?.call(value);
        return null;
      },
```